File: dart_lang/lib/b_behavioral/e_visitor/day_shift_schedule_management.dart
import 'package:dart_lang/b_behavioral/e_visitor/schedule_magngement.dart';
import 'package:dart_lang/b_behavioral/e_visitor/schedule_management_visitor.dart';

class DayShiftScheduleManagement implements ScheduleManagement {
  @override
  void accept(ScheduleManagementVisitor scheduleManagementVisitor) {
    scheduleManagementVisitor.visitDayShift(this);
  }

  @override
  void calcOverTime() {
    print(' Calculating over time for day shift employees');
  }

  @override
  void generateReport() {
    print('Generating report for day shift employees');
  }
}

File: PHP/Patterns/src/Behavioral/Visitor/DayShiftScheduleManagement.php
<?php

namespace src\Behavioral\Visitor;

use src\Behavioral\Visitor\Visitors\ScheduleManagementVisitor;

class DayShiftScheduleManagement implements ScheduleManagement
{
    function generateReport()
    {
        print_r('Generate report for day shift');
    }

    function calculateOverTime()
    {
        print_r('Calculate overtime for day shift');
    }

    function accept (ScheduleManagementVisitor $scheduleManagementVisitor) {
        $scheduleManagementVisitor->visitDayShift($this);
    }
}